mm2src/lp_ordermatch/simple_market_maker.rs

```diff
@@ -440,30 +521,60 @@ async fn prepare_order(
         .await?
         .ok_or_else(|| MmError::new(OrderProcessingError::AssetNotEnabled))?;
 
-    info!("balance for {} is {}", cfg.base, base_balance);
+    debug!("balance for {} is {}", cfg.base, base_balance);
 
     let mut calculated_price = rates.price * cfg.spread.clone();
-    info!("calculated price is: {}", calculated_price);
+    debug!("calculated price is: {}", calculated_price);
     if cfg.check_last_bidirectional_trade_thresh_hold.unwrap_or(false) {
-        calculated_price = vwap_calculator(calculated_price.clone(), ctx, &cfg).await?;
+        calculated_price = vwap_calculator(calculated_price.clone(), ctx, cfg).await?;
     }
 
-    let volume = match &cfg.balance_percent {
-        Some(balance_percent) => balance_percent * &base_balance,
-        None => MmNumber::default(),
+    let mut is_max = cfg.max.unwrap_or(false);
+
+    let volume = if let Some(max_volume) = &cfg.max_volume {
```

Comment:
```rust
match &cfg.max_volume {
  Some(VolumeSettings::Percentage(balance_percent)) => ...,
  Some(VolumeSettings::Usd(max_volume_usd)) => ...,
  None => ...
}
```

Is shorter and clearer.

Reply:
Thanks for the hints I was not aware it's was possible to do it this way!